### PR TITLE
[OpenVino] getcapability design

### PR DESCRIPTION
**Pull request by sfatimar**
Redesign of get capability to ensure that we remove the redundant capability files for each ov version

**Motivation and Context**
- This change is required to ensure that we follow a better design for getcapability
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/core/providers/openvino/ov_versions/capabilities.h [hunk 1]
```diff
@@ -2,26 +2,25 @@
...
+        std::string device_type_;
+        DataOps* data_ops_;
+    public:
+        GetCapability (const GraphViewer& graph_viewer_param, std::string device_type_param, const std::string version_param);   
+        virtual std::vector<std::unique_ptr<ComputeCapability>> Execute();
+};
+
 #if defined OPENVINO_2020_3
 std::vector<std::unique_ptr<ComputeCapability>>
 GetCapability_2020_3(const GraphViewer& graph_viewer, const std::string device_type);
```
Comment (HectorSVC) [inline onnxruntime/core/providers/openvino/ov_versions/capabilities.h:23]: why 2020.3 was not converted to this new method? are we going to deprecate it soon?

**Comment by HectorSVC** [comment]
@sfatimar, could you merge the latest master, the build failure looks wired.

**Review by sfatimar — COMMENTED** [review]
> @sfatimar, could you merge the latest master, the build failure looks wired.

Merged the master in

**Comment by HectorSVC** [comment]
@sfatimar There are some failed tests, please take a look.
1: [  FAILED  ] ModelTests/ModelTest.Run/446, where GetParam() = "openvino_../models/opset7/keras2coreml_BiDirectional_ImageNet/model.onnx"
1: [  FAILED  ] ModelTests/ModelTest.Run/753, where GetParam() = "openvino_../models/opset9/test_GPT2/model.onnx"
1: [  FAILED  ] ModelTests/ModelTest.Run/777, where GetParam() = "openvino_../models/opset10/GPT2/model.onnx"

**Comment by sfatimar** [comment]
@HectorSVC I am still not able to root cause this issue. When I run my build in internal CI I am not able to run the ModelTests.
When I try to run the gpt2 models from opset 10 and 11 repo I am not facing any issues with onnx_test_runner.
Will appreciate your help in this matter. Is there some way I can access these models and run them myself.

**Comment by HectorSVC** [comment]
1: [  FAILED  ] 1 test, listed below:
1: [  FAILED  ] ModelTests/ModelTest.Run/446, where GetParam() = "openvino_../models/opset7/keras2coreml_BiDirectional_ImageNet/model.onnx"
1: 
1:  1 FAILED TEST
1:   YOU HAVE 4 DISABLED TESTS
1: 
1/6 Test #1: onnxruntime_test_all ...................***Failed  1617.48 sec

**Comment by sfatimar** [comment]
Resolved build issues

**Review by HectorSVC — APPROVED** [review]
:shipit:

**Pull request merged**